llrt_core/src/modules/http/response.rs

```diff
@@ -149,7 +166,21 @@ impl<'js> Response<'js> {
                     body.body_mut().collect().await.or_throw(ctx)?.to_bytes()
                 };
 
-                bytes.to_vec()
+                if self.body_attributes.content_encoding.is_none() {
+                    return Ok(Some(bytes.to_vec()));
+                }
+
+                let content_encoding = self.body_attributes.content_encoding.clone();
+                let mut data: Vec<u8> = Vec::new();
```

Comment:
How will the decoders below allocate? Do they reserve based on max size or does it append the the vec? Would it make sense to set `Vec::with_capactity` to at least since of bytes to reduce allocations?

Reply:
I was wondering what to do about pre-allocating vectors, too.

Checking the `std::io::read_to_end()` implementation (actually `std::io::default_read_to_end()`), it looks like it does `Vec::try_reserve()` by 32 bytes if there is insufficient capacity. So, as you say, it would be effective to allocate in advance.

- std::io::read_to_end()`:  https://doc.rust-lang.org/src/std/io/mod.rs.html#851-854
- std::io::default_read_to_end()`:  https://doc.rust-lang.org/src/std/io/mod.rs.html#410-523

The best thing to do would have been to reserve the capacity in advance with `content-length`, but the response returned in chunked format is not given `content-length`, and it does not seem to be possible to know the just size in advance.

An idea to minimize the additional allocation as much as possible is to reserve "roughly twice" the size of the response body in advance, for example, and `Vec::shrink_to_fit()` it later.

What do you think?

Reply:
I think we can start to reserve at least the size of the compressed bytes

Reply:
I will go with that idea.